api/oc_push.c

```diff
@@ -0,0 +1,2605 @@
+				 * update URI of Resource pointed by previous `receiveruri` */
+				if (strcmp(oc_string(recv_obj->receiveruri), oc_string(rep->value.string)))
+				{
+					pushd_rsc_rep = _find_pushd_rsc_rep_by_uri(&recv_obj->receiveruri, recvs_instance->resource->device);
+
+					if (pushd_rsc_rep)
+					{
+						OC_PUSH_DBG("pushed resource representation (\"%s\") is found", oc_string(pushd_rsc_rep->resource->uri));
+
+						oc_free_string(&pushd_rsc_rep->resource->uri);
+						oc_store_uri(oc_string(rep->value.string), &pushd_rsc_rep->resource->uri);
+					}
+
+				}
+
+				oc_free_string(&recv_obj->receiveruri);
+				oc_new_string(&recv_obj->receiveruri, oc_string(rep->value.string), oc_string_len(rep->value.string));
+			}
+			break;
+
+		case OC_REP_STRING_ARRAY:
+			if (!strcmp(oc_string(rep->name), "rts"))
+			{
+				oc_free_string_array(&recv_obj->rts);
+				int len = oc_string_array_get_allocated_size(rep->value.array);
+				oc_new_string_array(&recv_obj->rts, len);
+
+				for (int i=0; i < len; i++)
+				{
+					oc_string_array_add_item(recv_obj->rts, oc_string_array_get_item(rep->value.array, i));
+				}
+			}
+			break;
+
+		default:
+			OC_PUSH_ERR("something wrong, unexpected Property type: %d", rep->type);
+			return;
+		}
+		rep = rep->next;
+	}
+
+	return;
+}
+
+
+
+/*
+ * @brief				create new receiver object
+ *
+ * @param recv_obj	pointer for new receiver object
+ * @param rep			received new receiver object object
+ */
+static bool _create_recv_obj(oc_recvs_t *recvs_instance, oc_rep_t *rep)
+{
+	bool result = false;
+	char mandatory_property_check = 0;
+	oc_recv_t *recv_obj = (oc_recv_t *)oc_memb_alloc(&recv_instance_memb);
+
+	if (!recv_obj)
+	{
+		OC_PUSH_ERR("oc_memb_alloc() error!");
+		return result;
+	}
+
+	while (rep)
+	{
+		switch (rep->type)
+		{
+		case OC_REP_STRING:
+			if (!strcmp(oc_string(rep->name), "receiveruri"))
+			{
+				oc_new_string(&recv_obj->receiveruri, oc_string(rep->value.string), oc_string_len(rep->value.string));
+				mandatory_property_check |= 0x1;
+			}
+			break;
+
+		case OC_REP_STRING_ARRAY:
+			if (!strcmp(oc_string(rep->name), "rts"))
+			{
+				int len = oc_string_array_get_allocated_size(rep->value.array);
+				oc_new_string_array(&recv_obj->rts, len);
+
+				for (int i=0; i < len; i++)
+				{
+					oc_string_array_add_item(recv_obj->rts, oc_string_array_get_item(rep->value.array, i));
+				}
+
+				mandatory_property_check |= 0x2;
+			}
+			break;
+
+		default:
+			OC_PUSH_ERR("something wrong, unexpected Property type: %d", rep->type);
+			break;
+		}
+		rep = rep->next;
+	}
+
+	if (mandatory_property_check != 0x3)
+		return result;
+	else
+		result = true;
+
+	oc_list_add(recvs_instance->receivers, recv_obj);
+
+	/* create app resource corresponding to receiver object */
+	OC_PUSH_DBG("new app resource for new receiver obj (\"%s\") is created...", oc_string(recv_obj->receiveruri));
+	if (_create_pushd_rsc(recv_obj, recvs_instance->resource))
+		result = true;
```

Comment:
Isn't this inverted? You either return on line 1900 or go to the else branch, which sets `result=true`. So the value is already set to `true`. Maybe you should set the result to `false` in case it fails and log error?